### PR TITLE
Fix bug where admin can read system index

### PR DESCRIPTION
**Pull request by derek-ho**
### Description
Prevents reads on security system index registered through core

### Issues Resolved
Fix: #4755 
Is this a backport? If so, please add backport PR # and/or commits #, and remove `backport-failed` label from the original PR.
No
Do these changes introduce new permission(s) to be displayed in the static dropdown on the front-end? If so, please open a draft PR in the security dashboards plugin and link the draft PR here
No
### Testing
Added unit test
### Check List
- [x] New functionality includes testing
- [ ] New functionality has been documented
- [ ] New Roles/Permissions have a corresponding security dashboards plugin PR
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md)
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/security/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/integrationTest/java/org/opensearch/security/SystemIndexTests.java [hunk 1]
```diff
@@ -80,4 +80,29 @@ public void adminShouldNotBeAbleToDeleteSecurityIndex() {
...
+            assertThat(response2.getStatusCode(), equalTo(RestStatus.OK.getStatus()));
+            assertThat(response2.getBody(), response2.getBody().contains("\"hits\":{\"total\":{\"value\":1,\"relation\":\"eq\"}"));
+        }
+
+        // Regular users should not be able to read it
+        try (TestRestClient client = cluster.getRestClient(USER_ADMIN)) {
+            // regular user cannot read system index
+            HttpResponse response1 = client.get(".system-index1/_search");
+
+            assertThat(response1.getBody(), response1.getBody().contains("\"hits\":{\"total\":{\"value\":0,\"relation\":\"eq\"}"));
```
Comment (derek-ho) [inline src/integrationTest/java/org/opensearch/security/SystemIndexTests.java:105]: Verified this line fails without the change.

File: src/main/java/org/opensearch/security/configuration/SecurityIndexSearcherWrapper.java [hunk 2]
```diff
@@ -39,6 +39,7 @@
 import org.opensearch.core.common.transport.TransportAddress;
```
Comment (cwperks) [inline src/main/java/org/opensearch/security/configuration/SecurityIndexSearcherWrapper.java:39]: Can we rename this class to SystemIndexSearcherWrapper?

File: src/integrationTest/java/org/opensearch/security/SystemIndexTests.java [hunk 3]
```diff
@@ -80,4 +80,29 @@ public void adminShouldNotBeAbleToDeleteSecurityIndex() {
             assertThat(response4.getStatusCode(), equalTo(RestStatus.FORBIDDEN.getStatus()));
         }
     }
+
+    @Test
+    public void adminShouldNotBeAbleToReadSecurityIndex() {
```
Comment (cwperks) [inline src/integrationTest/java/org/opensearch/security/SystemIndexTests.java:85]:
```suggestion
    public void adminShouldNotBeAbleToReadSystemIndex() {
```

File: src/main/java/org/opensearch/security/configuration/SecurityIndexSearcherWrapper.java (hunk 2 above)
Comment (derek-ho) [reply src/main/java/org/opensearch/security/configuration/SecurityIndexSearcherWrapper.java:39]: Done in: [7402adb](https://github.com/opensearch-project/security/pull/4774/commits/7402adb5f16740e05cabfc211534e13a24985ac9)

File: src/integrationTest/java/org/opensearch/security/SystemIndexTests.java (hunk 3 above)
Comment (derek-ho) [reply src/integrationTest/java/org/opensearch/security/SystemIndexTests.java]: Done in: [7459bc2](https://github.com/opensearch-project/security/pull/4774/commits/7459bc211e0281c9cd1266347b4c755915606e2e)

**Review by willyborankin — APPROVED** [review]

**Review by cwperks — APPROVED** [review]

**Pull request merged**